addons/purchase/models/purchase_order.py

```diff
@@ -537,14 +542,13 @@ def _add_supplier_to_product(self):
             already_seller = (partner | self.partner_id) & line.product_id.seller_ids.mapped('partner_id')
             if line.product_id and not already_seller and len(line.product_id.seller_ids) <= 10:
                 # Convert the price in the right currency.
-                currency = partner.property_purchase_currency_id or self.env.company.currency_id
-                price = self.currency_id._convert(line.price_unit, currency, line.company_id, line.date_order or fields.Date.today(), round=False)
+                price = line.price_unit
```

Comment:
The comment `# Convert the price in the right currency.` is not relevant anymore